### PR TITLE
Add fs2 module

### PR DESCRIPTION
**Pull request by keirlawson**

### CODE REVIEW

File: modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package pureconfig.module.fs2
```
Comment (ruippeixotog) [inline modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala]: Nitpick: this file should be in `modules/fs2/src/test/scala/pureconfig/module/fs2/fs2Suite.scala` (with a fs2 subfolder) in order to match the package name.

File: modules/fs2/build.sbt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "co.fs2" %% "fs2-io" % "0.10.2",
+  Dependencies.scalaMacrosParadise,
+  Dependencies.scalaTest)
+
+// fs2 0.10 isn't published for Scala 2.10
+crossScalaVersions ~= { oldVersions => oldVersions.filterNot(_.startsWith("2.10")) }
+
+pomExtra := {
+  <scm>
+    <url>git@github.com:pureconfig/pureconfig.git</url>
```
Comment (ruippeixotog) [inline modules/fs2/build.sbt]: You shouldn't specify the SCM information here. Doing that clashes with `sbt-git`, which also injects SCM information in the POM file, leading to invalid (duplicate) SCM entries.

**Review by ruippeixotog — CHANGES_REQUESTED** [review]
LGTM overall, thanks for the new module!

File: modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala (hunk 1 above)
Comment (keirlawson) [reply modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala]: I feel that is is more important to mirror the src tree, which is `modules/fs2/src/main/scala/pureconfig/module/fs2.scala` as it is a package object, however it is pretty minor so if you insist on it I'm happy to change :)

File: modules/fs2/build.sbt (hunk 2 above)
Comment (keirlawson) [reply modules/fs2/build.sbt]: Fixed, not sure how that slipped in!

File: modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala (hunk 1 above)
Comment (ruippeixotog) [reply modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala]: Sure, it makes all sense to mirror the source tree 👍 But then this package should be `pureconfig.module` just like the `fs2` package object has, isn't it?

**Comment by keirlawson** [comment]
Changes made to address feedback

**Review by ruippeixotog — APPROVED** [review]
LGTM!

**Review by leifwickland — APPROVED** [review]
Thanks for the addition, @keirlawson!

**Pull request merged**